lib/src/models/comment_impl.dart

```diff
@@ -99,13 +111,17 @@ class MoreComments extends RedditBase {
   }
 
   Future<Comment> _loadComment(String commentId) async {
+    if (_submission is! Submission) {
+      _submission = await _submission.populate();
+    }
+    final submissionCast = _submission as Submission;
```

Comment:
What kind of voodoo is this?

Reply:
Basically, `_submission` is a `SubmissionRef` so what we do here is ensure that `_submission` is populated. If it's not, `_submission` is an actual instance of `SubmissionRef` so we can populate it and get a `Submission` back. 

This is bad style on my part, I should have done this instead:

```dart
final Submission submissionCast = _submission;
```

Which accomplishes the same thing. If we were in C++, this is the same as me doing a cast to a `Submission` object.